client/matching/client_test.go

```diff
@@ -0,0 +1,521 @@
+			},
+			want: &types.ListTaskListPartitionsResponse{},
+		},
+		{
+			name: "ListTaskListPartitions - Error in resolving peer",
+			op: func(c Client) (any, error) {
+				return c.ListTaskListPartitions(context.Background(), testMatchingListTaskListPartitionsRequest())
+			},
+			mock: func(p *MockPeerResolver, balancer *MockLoadBalancer, c *MockClient) {
+				p.EXPECT().FromTaskList(_testTaskList).Return("peer0", assert.AnError)
+			},
+			want:      nil,
+			wantError: true,
+		},
+		{
+			name: "ListTaskListPartitions - Error while listing tasklist partitions",
+			op: func(c Client) (any, error) {
+				return c.ListTaskListPartitions(context.Background(), testMatchingListTaskListPartitionsRequest())
+			},
+			mock: func(p *MockPeerResolver, balancer *MockLoadBalancer, c *MockClient) {
+				p.EXPECT().FromTaskList(_testTaskList).Return("peer0", nil)
+				c.EXPECT().ListTaskListPartitions(gomock.Any(), gomock.Any(), []yarpc.CallOption{yarpc.WithShardKey("peer0")}).Return(nil, assert.AnError)
+			},
+			want:      nil,
+			wantError: true,
+		},
+		{
+			name: "GetTaskListsByDomain",
+			op: func(c Client) (any, error) {
+				return c.GetTaskListsByDomain(context.Background(), testGetTaskListsByDomainRequest())
+			},
+			mock: func(p *MockPeerResolver, balancer *MockLoadBalancer, c *MockClient) {
+				p.EXPECT().GetAllPeers().Return([]string{"peer0", "peer1"}, nil)
+				c.EXPECT().GetTaskListsByDomain(gomock.Any(), testGetTaskListsByDomainRequest(), []yarpc.CallOption{yarpc.WithShardKey("peer0")}).Return(&types.GetTaskListsByDomainResponse{}, nil)
+				c.EXPECT().GetTaskListsByDomain(gomock.Any(), testGetTaskListsByDomainRequest(), []yarpc.CallOption{yarpc.WithShardKey("peer1")}).Return(&types.GetTaskListsByDomainResponse{}, nil)
+			},
+			want: &types.GetTaskListsByDomainResponse{
+				DecisionTaskListMap: make(map[string]*types.DescribeTaskListResponse),
+				ActivityTaskListMap: make(map[string]*types.DescribeTaskListResponse),
+			},
+		},
+		{
+			name: "GetTaskListsByDomain - Error in resolving peer",
+			op: func(c Client) (any, error) {
+				return c.GetTaskListsByDomain(context.Background(), testGetTaskListsByDomainRequest())
+			},
+			mock: func(p *MockPeerResolver, balancer *MockLoadBalancer, c *MockClient) {
+				p.EXPECT().GetAllPeers().Return([]string{"peer0", "peer1"}, assert.AnError)
+			},
+			want:      nil,
+			wantError: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+
+			// setting up client
+			ctrl := gomock.NewController(t)
+			client := NewMockClient(ctrl)
+			peerResolverMock := NewMockPeerResolver(ctrl)
+			loadbalancerMock := NewMockLoadBalancer(ctrl)
+			tt.mock(peerResolverMock, loadbalancerMock, client)
+			c := NewClient(client, peerResolverMock, loadbalancerMock)
+
+			res, err := tt.op(c)
+			if tt.wantError {
+				assert.Error(t, err)
+				assert.Nil(t, res)
```

Comment:
Nit: I think we should not care about the result when we return an error

Reply:
I agree to some extent but also think its good to keep this check to make sure all results are included